src/compas_fab/utilities/file_io.py

```diff
@@ -14,11 +14,22 @@
 def read_csv_to_dictionary(csvfile, delimiter=';'):
     """Reads a csv file and returns a dictionary with the respective keys
     specified in the first row of the csv file.
+
+    Parameters
+    ----------
+    csvfile : str
+        The path to csv file.
+    delimiter : str, optional
+        The character used to separate the values. Default `;`
+
+    Returns
+    -------
+    dict
     """
     data = []
     with open(csvfile, mode='r') as infile:
         reader = csv.reader(infile, delimiter=delimiter)
-        for i, rows in enumerate(reader):
+        for _i, rows in enumerate(reader):
```

Comment:
Very tidy.

Reply:
why do you need the enumerate in the first place?

Reply:
I think you don't.  It could be changed to `for rows in reader:`

Reply:
https://docs.python.org/3/library/csv.html#csv.reader